### DISCUSSION TITLE
Quick question about JSON, TOML & YAML tags

### DISCUSSION THREAD

**Discussion by treddy73** [aarondl/sqlboiler#919] [General]
I'm learning Go and SQLBoiler and I have a pretty naive question about the tags added to the DB model structs.  Here's what I'm referring to:
```
ID      int    `boil:"id" json:"id" toml:"id" yaml:"id"`
```
I understand why there's a `boil` tag, but in what circumstances would we wish to marshal this struct and serialize any other way?  Even if, say, an API endpoint returns a JSON response that is identical in all respects to the underlying DB table-model, it still seems like a good idea to separate these domains (front-end request/response and back-end table-model) into separate structs.

Is there some "reflection-like" API that will programmagically map one struct to another using these tags?

**Comment by aarondl** [comment]
It's true there's likely going to be a separate object for any api you have. These struct tags can be useful for smaller applications that just want to do things quickly.

As for reflect based struct mapping you'll have to look outside this project. Personally I've written my own in the past but none are open source atm.
